tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,964 @@
+    # Fetch contents of tables files_qcstatus #
+    # and feedback_mri_comments               #
+    #=========================================#
+    (my $query =<<QUERY) =~ s/\s+/ /g;
+         SELECT fqs.FileID
+         FROM files_qcstatus fqs 
+         JOIN files f USING (FileID)
+         WHERE f.TarchiveSource = ?
+         
+         UNION
+          
+         SELECT fmc.FileID
+         FROM feedback_mri_comments fmc 
+         JOIN files f USING (FileID)
+         WHERE f.TarchiveSource = ?
+         LIMIT 1
+QUERY
+
+    my $rowsRef = $dbh->selectall_arrayref($query, undef, $tarchiveID, $tarchiveID);
+
+    return @$rowsRef > 0;
+}
+
+=pod
+
+=head3 getFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Get the absolute paths of all the files associated to an archive that are listed in 
+table C<files>.
+
+INPUTS:
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+
+RETURNS: 
+ - an array of hash references. Each hash has three keys: C<FileID> => ID of a file in table C<files>
+   C<File> => value of column C<File> for the file with the given ID and C<FullPath> => absolute path
+   for the file with the given ID.
+
+=cut
+sub getFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # Get FileID and File path of each files in files directly tied
+    # to $tarchiveId
+    $query = 'SELECT FileID, File FROM files WHERE TarchiveSource = ?';
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Set full path of every file
+    foreach(@$filesRef) {
+        $_->{'FullPath'} = $_->{'File'} =~ /^\// 
+            ? $_->{'File'} : "$dataDirBasePath/$_->{'File'}";
+    } 
+    
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getIntermediaryFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Get the absolute paths of all the intermediary files associated to an archive 
+that are listed in table C<files_intermediary>.
+
+INPUTS:
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+
+RETURNS: 
+  - an array of hash references. Each hash has four keys: C<IntermedID> => ID of a file in 
+  table C<files_intermediary>, C<FileID> => ID of this file in table C<files>, C<File> => value
+  of column C<File> in table C<files> for the file with the given ID and C<FullPath> => absolute
+  path of the file with the given ID.
+
+=cut
+sub getIntermediaryFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # This should get all files in table files_intermediary that are tied
+    # indirectly to the tarchive with ID $tarchiveId
+    my $query = 'SELECT fi.IntermedID, f.FileID, f.File FROM files_intermediary fi '
+        .       'JOIN files f ON (fi.Output_FileID=f.FileID) '
+        .       'WHERE f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)';
```

Comment:
add a `GROUP BY FileID` to the query so that the files are not being duplicated in that query since there could be multiple entries in `files_intermediary` with the same `FileID`. I think that would solve the issue I had with:
```
Warning! File '/data/preventAD/data/assembly/857385/NAPEN00/mri/processed/DTIPrepPipeline/PreventAD_857385_NAPEN00_dti_001_DTIPrepBaseline_001.mnc' could not be deleted: No such file or directory
Warning! File '/data/preventAD/data/assembly/857385/NAPEN00/mri/processed/DTIPrepPipeline/PreventAD_857385_NAPEN00_dti_001_DTIPrepBetMask_001.mnc' could not be deleted: No such file or directory
Warning! File '/data/preventAD/data/assembly/857385/NAPEN00/mri/processed/DTIPrepPipeline/PreventAD_857385_NAPEN00_dti_001_DTIPrepDTIColorFA_001.mnc' could not be deleted: No such file or directory
Warning! File '/data/preventAD/data/assembly/857385/NAPEN00/mri/processed/DTIPrepPipeline/PreventAD_857385_NAPEN00_dti_001_DTIPrepDTIMD_001.mnc' could not be deleted: No such file or directory
Warning! File '/data/preventAD/data/assembly/857385/NAPEN00/mri/processed/DTIPrepPipeline/PreventAD_857385_NAPEN00_dti_001_DTIPrepDTIFA_001.mnc' could not be deleted: No such file or directory
```
(note, they were all processed files with multiple entries in files_intermediary